### PR TITLE
[AIR] Stabilize MosaicTrainer tests

### LINKED ISSUE #29769: [CI] `linux://python/ray/train:test_mosaic_trainer` is flaky
image

```
python/ray/train/tests/test_mosaic_trainer.py::test_mosaic_cifar10 -- Test timed out at 2022-10-26 01:20:56 UTC --

...

  | //python/ray/train:test_mosaic_trainer                                    FLAKY, failed in 1 out of 2 in 302.6s
Stats over 2 runs: max = 302.6s, min = 61.6s, avg = 182.1s, dev = 120.5s
```

### PR DESCRIPTION
**Pull request by ilee300a**
Increased the training epoch to stabilize the accuracy check for `test_mosaic_cifar10` and removed `prepare_model` wrapping to avoid `test_mosaic_trainerpy` hanging.

Closes https://github.com/ray-project/ray/issues/29769

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by amogkam** [comment]
Looks like this is still timing out during the first run 

```
//python/ray/train:test_mosaic_trainer                                    FLAKY, failed in 1 out of 2 in 303.0s
--
  | Stats over 2 runs: max = 303.0s, min = 164.9s, avg = 233.9s, dev = 69.1s
  | /root/.cache/bazel/_bazel_root/5fe90af4e7d1ed9fcf52f59e39e126f5/execroot/com_github_ray_project_ray/bazel-out/k8-opt/testlogs/python/ray/train/test_mosaic_trainer/test_attempts/attempt_1.log
```

**Comment by jiaodong** [comment]
change lgtm but can you try coming up with a rough breakdown of where time was spent for all tests and which test takes longest time ? Maybe there're a few simple fixes we can do to ensure it always passes under time constraint.

File: python/ray/train/examples/mosaic_cifar10_example.py [hunk 1]
```diff
@@ -82,7 +81,7 @@ def train_mosaic_cifar10(num_workers=2, use_gpu=False):
     from ray.train.mosaic import MosaicTrainer
 
     trainer_init_config = {
-        "max_duration": "2ep",
+        "max_duration": "5ep",
```
Comment (bveeramani) [inline python/ray/train/examples/mosaic_cifar10_example.py:84]: Nit: `max_duration` should be a `train_mosaic_cifar10` parameter. 

Readers won't be able to understand why
```
 assert result["epoch"][result.index[-1]] == 4
```
unless they look at the implementation for `train_mosaic_cifar10`.

File: python/ray/train/tests/test_mosaic_trainer.py [hunk 2]
```diff
@@ -88,10 +86,10 @@ def test_mosaic_cifar10(ray_start_4_cpus):
     result = train_mosaic_cifar10().metrics_dataframe
```
Comment (bveeramani) [inline python/ray/train/tests/test_mosaic_trainer.py:86]: Nit: See comment above.

```suggestion
    result = train_mosaic_cifar10(max_duration="5ep").metrics_dataframe
```

This'd make the test easier to understand.

File: python/ray/train/tests/test_mosaic_trainer.py [hunk 3]
```diff
@@ -22,10 +21,9 @@ def trainer_init_per_worker(config):
     import composer.optim
 
     BATCH_SIZE = 32
-    # prepare the model for distributed training and wrap with ComposerClassifier for
-    # Composer Trainer compatibility
-    model = config.pop("model", torchvision.models.resnet18(num_classes=10))
-    model = ComposerClassifier(ray.train.torch.prepare_model(model))
+    model = ComposerClassifier(
+        config.pop("model", torchvision.models.resnet18(num_classes=10))
```
Comment (bveeramani) [inline python/ray/train/tests/test_mosaic_trainer.py:25]: If we don't call `prepare_model`, does Mosaic still perform distributed training? 

IIRC if you don't call `prepare_model` in normal Torch, the program trains `num_workers` independent models (as opposed to one distributed model).

Comment (ilee300a) [reply python/ray/train/tests/test_mosaic_trainer.py:25]: Composer library automatically checks the world size and rank and applies distributed training

**Review by jiaodong — APPROVED** [review]
`//python/ray/train:test_mosaic_trainer                                   PASSED in 151.3s`
 
this is looking much faster than before :)

**Pull request merged**